frontend/packages/dev-console/src/components/import/GitImportForm.tsx

```diff
@@ -40,7 +40,9 @@ const GitImportForm: React.FC<FormikProps<FormikValues> & GitImportFormProps> =
       isSubmitting={isSubmitting}
       submitLabel="Create"
       sticky
-      disableSubmit={!dirty || !_.isEmpty(errors)}
+      disableSubmit={
+        values.git.urlValidation === ValidatedOptions.default || !dirty || !_.isEmpty(errors)
+      }
```

Comment:
Why do you need to do this when you are adding the validation that will add the error if repo url is not correct? https://github.com/openshift/console/pull/5145/files#diff-ad8be16d16eef75fed4951b2be470be8R234-R244. If the error is there the button will anyways get disabled.

Reply:
validation will occur only if the user clicks outside the input field. So there can be a scenario where all the required fields are filled and the user goes ahead and edits just the url. The create button is NOT disabled when the user makes changes to the url. As a result, the user can very easily click on the Create button without validating the git url. Below is the gif to give you a clear picture:

![gitimport](https://user-images.githubusercontent.com/22490998/80069503-15a8e900-855f-11ea-8d52-39e6f118c975.gif)

I felt that we should be taking care of such cases. Please let me know if you would like to handle this in a different way

Reply:
Well, we do have a bug that will change the `onBlur` behavior to `onChange`. That will validate the URL while user is typing so you won't the explicit check like this. But for now I would say its okay to have this change.

Reply:
@rohitkrai03 you must be talking about this issue https://issues.redhat.com/browse/ODC-2451, where the validation should take place right away as the user types the url

Reply:
Right.

Reply:
The above gif you showed has an inherent problem itself. Validations run every time user clicks on the submit button and we can see that it shows the error after that. So why does the submit handler submits the form even when there is an error?

Reply:
The submit action was called before the url could be validated and this was possible only because the Submit button was not disabled.

Reply:
Submit handler waits for the validation to complete. Formik runs the validation again before calling onSubmit handler https://github.com/jaredpalmer/formik/blob/master/packages/formik/src/Formik.tsx#L740. But the problem is git service validations run async and separate from form validation schema and there is no way of checking errors in the `onSubmit` handler since formik omits that in `formikBag`

Reply:
Yes. This I did mention in the Root cause, in the PR summary, that the url validation that we are doing, if errors occur it doesn't show up in the formik errors.